System/Directory/Internal/Posix/Template.hs

```diff
@@ -287,40 +257,40 @@ setFileTimes' pth atime' mtime' =
 #endif
 
 -- | Get the contents of the @PATH@ environment variable.
-getPath :: IO [FilePath]
-getPath = splitSearchPath <$> getEnv "PATH"
+getPath :: IO [FILEPATH]
+getPath = splitSearchPath <$> getEnv (fromString "PATH")
 
 -- | $HOME is preferred, because the user has control over it. However, POSIX
 -- doesn't define it as a mandatory variable, so fall back to `getpwuid_r`.
-getHomeDirectoryInternal :: IO FilePath
+getHomeDirectoryInternal :: IO FILEPATH
 getHomeDirectoryInternal = do
-  e <- lookupEnv "HOME"
+  e <- lookupEnv (fromString "HOME")
   case e of
        Just fp -> pure fp
-       Nothing -> PU.homeDirectory <$> (PU.getEffectiveUserID >>= PU.getUserEntryForID)
+       -- TODO: fromString here is bad, but unix's System.Posix.User.UserEntry does not have ByteString/OsString variants
```

Comment:
don't see another way currently

Reply:
It would break API, because the type has `String` hardcoded in its records, so I'm not sure yet what to do.